config/externalname.go

```diff
@@ -2989,6 +2994,12 @@ func getPermissionSetId(tfstate map[string]any) (string, error) {
 // IDs that use elements from the parameters in a certain string format.
 // It should be used in cases where all information in the ID is gathered from
 // the spec and not user defined like name. For example, zone_id:vpc_id.
+//
+// TODO: This should set keys as IdentifierFields, because if they're missing observe-only resources won't work.
+// But that would remove them from spec.initProvider, which would be a breaking schema change for existing resources
+// that we don't have a good way to handle yet.
+//
+// For new resources, prefer using FormattedIdentifierFromProviderWithIdentifierFields instead.
```

Comment:
@ulucinar I'd love to come up with a better solution for the (many) cases where we have resources with fields that should be identifier fields, but are not. The options I can think of are:
1. Make a way to include them in both `spec.forProvider` and `spec.initProvider`.
2. Remove immutable fields from `spec.initProvider` in a new api version
3. Remove the need for them to be required by allowing the terraform id to be constructed from either the parameters or the external name.

I think the last would probably be the best long-term user experience, but I don't know how hard it is, or if it would cause other problems. 

What do you think?

Reply:
I'll put it on the sig-upjet agenda doc.

Reply:
https://github.com/crossplane/upjet/issues/422